File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/util/BinarySearch.java
package org.firstinspires.ftc.teamcode.util;

public class BinarySearch {

    private BinarySearch() {}

    /**
     * Returns the index of the first value >= target.
     * Assumes values are sorted in ascending order.
     */
    public static int firstGreaterOrEqual(double[] values, double target) {

        int low = 0;
        int high = values.length - 1;

        while (low < high) {
            int mid = (low + high) >>> 1;

            if (values[mid] >= target)
                high = mid;
            else
                low = mid + 1;
        }

        return low;
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/util/BinarySearch.kt
package org.firstinspires.ftc.teamcode.util

object BinarySearch {

    fun firstGreaterOrEqual(values: DoubleArray, target: Double): Int {

        var low = 0
        var high = values.lastIndex

        while (low < high) {
            val mid = (low + high) ushr 1
            if (values[mid] >= target) high = mid
            else low = mid + 1
        }

        return low
    }
}